integration_test/integration_test.go

```diff
@@ -72,7 +72,7 @@ func startTCPEchoServer(t testing.TB) (*net.TCPListener, *sync.WaitGroup) {
 }
 
 func startUDPEchoServer(t testing.TB) (*net.UDPConn, *sync.WaitGroup) {
```

Comment:
It would be great to exercise the new behavior somehow, so we can be confident it's not breaking things.
Also add IPv6 testing.

Reply:
For testing, perhaps you can listen on two separate IPs, like 127.0.0.1 and 127.0.0.2?

Reply:
I was able to test using 127.0.0.1 and 127.0.0.2, but I can't easily write IPv6 tests because (1) IPv6 doesn't have an analogue to 127.0.0.2 and (2) my workstation (and presumably some other potential tests environments) has no IPv6 support.